libnd4j/tests_cpu/layers_tests/SparseUtilsTest.cpp

```diff
@@ -144,4 +144,95 @@ TEST_F(SparseUtilsTest, SortCOOindices_Test) {
     delete[] expIndicesArr;
 
     #endif
+}
+
+//////////////////////////////////////////////////////////////////////
+TEST_F(SparseUtilsTest, RavelIndices_Test) {
+    Nd4jLong * indicesArrExp = new Nd4jLong[nnz * rank]{
+            0,2,7,
+            2,36,35,
+            3,30,17,
+            5,12,22,
+            5,43,45,
+            6,32,11,
+            8,8,32,
+            9,29,11,
+            5,11,22,
+            15,26,16,
+            17,48,49,
+            24,28,31,
+            26,6,23,
+            31,21,31,
+            35,46,45,
+            37,13,14,
+            6,38,18,
+            7,28,20,
+            8,29,39,
+            8,32,30,
+            9,42,43,
+            11,15,18,
+            13,18,45,
+            29,26,39,
+            30,8,25,
+            42,31,24,
+            28,33,5,
+            31,27,1,
+            35,43,26,
+            36,8,37,
+            39,22,14,
+            39,24,42,
+            42,48,2,
+            43,26,48,
+            44,23,49,
+            45,18,34,
+            46,28,5,
+            46,32,17,
+            48,34,44,
+            49,38,39,
+    };
+    Nd4jLong * indicesArr = new Nd4jLong[nnz * rank];
+
+        Nd4jLong * flatIndicesExp = new Nd4jLong[nnz]{
+            147,  10955,  14717,  21862,  24055,  27451,  34192,  39841,
+            21792,  64836,  74809, 102791, 109643, 131701, 150265, 156324,
+            27878,  31380,  35669,  35870,  40783,  47268,  55905, 123659,
+            126585, 178594, 119915, 132091, 150036, 151797, 165354, 165522,
+            179762, 182468, 186459, 190294, 195165, 195457, 204024, 208499
+    };
+
+    Nd4jLong * flatIndices = new Nd4jLong[nnz];
+
+
+    Nd4jLong * shape =  new Nd4jLong[rank]{50, 60, 70};
+    Nd4jLong * shapeInfoBuffer = shape::shapeBuffer(rank, shape);
```

Comment:
this one gives build failure. 
`Nd4jLong *shapeBuffer(int rank, sd::DataType dtype, Nd4jLong const* shape);`